### DISCUSSION TITLE
OIDC token settings

### DISCUSSION THREAD

**Discussion by cuttingedge1109** [keycloak/keycloak#13462] [Q&A]
- Which setting is for refresh token lifespan? I can see there is `Access Token Lifespan` setting in `Tokens` setting but cannot find one for refresh token.

- Which one has higher priority? I can configure token settings in realm setting and also in client settings.

- How can i revoke access tokens issued already using Not-Before-Policy function? Please check my understanding on Not-Before-Policy. 
I have to realize an endpoint on the client side which receives not-before push notification from keycloak. And i have to decrypt access token every time and check if isseAt is newer than the not-before time. If it is older, the token is considered as invalid. 
This logic should be realized on client side by myself. Am i right?

**Comment by sschu** [comment]
Hi @cuttingedge1109,

- refresh token settings in Keycloak are tied to SSO settings, so refresh token lifetime equals SSO session idle time with an upper bound at SSO session max.
- client settings override realm settings for a particular client
- I am not 100% sure Keycloak supports this so I cannot answer this question.

**Comment by bonnm** [comment]
An addition: If you want to decouple refresh token lifespans from SSO (cookie lifetime) settings on realm-level, you can do this by setting the Client [Offline] Session Idle/Max values to values higher than zero.
